### PR TITLE
add nitrokey3 update in qubes

### PR DESCRIPTION
**Pull request by nestire**

### CODE REVIEW

**Review by daringer — APPROVED** [review]

File: nitrokey3/linux/firmware-update-qubes.rst [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+How to Update
+-------------
+
+.. important::
+   Never disconnect the Nitrokey 3 or abort the process while updating,
+   this will likely render your device useless!
+
+1. Run ``nitropy nk3 update`` in sys-usb.
+2. Once instructed by ``nitropy`` touch the device to activate bootloader.
+3. Please wait until the process finished. (This may vary depending on your operating system)
```
Comment (jans23) [inline nitrokey3/linux/firmware-update-qubes.rst:61]: Here the OS is Qubes OS so this sentence doesn't make any sense.

File: nitrokey3/linux/firmware-update-qubes.rst [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+4. *Optional*: run ``nitropy nk3 test`` to check if device is working properly after flashing.
+5. Restart sys-usb and disable the network in Qubes Settings again
+
+In case of any errors please take the logs from ``/tmp`` directory (``/tmp/nitropy.log.*``).
+
+.. _Nitrokey 3 Firmware - GitHub Releases: https://github.com/Nitrokey/nitrokey-3-firmware/releases
+.. _installation instructions: ../../software/nitropy/all-platforms/installation.html
+
+
+Firmware Release Types
```
Comment (jans23) [inline nitrokey3/linux/firmware-update-qubes.rst:71]: Don't copy text from other files but include it instead or refer to it.

**Comment by jans23** [comment]
Images don't belong to the "fido2" folder so move it to "nitrokey3" folder.

**Comment by tlaurion** [comment]
@daringer please update so the documentation can be tested in goal of closing https://github.com/Nitrokey/nitrokey-app2/issues/249 and then merging https://github.com/linuxboot/heads/pull/1684

**Comment by tlaurion** [comment]
Dodging sys-usb requirement and redoing under debian-12 test qube, where sys-usb should never have network access and where passing device through device widget once mode is switched to flashing should mitigate the problem altogether.

sudo apt install python3-pip wget
python3 -m pipx ensurepath
wget https://raw.githubusercontent.com/Nitrokey/libnitrokey/master/data/41-nitrokey.rules
sudo mv 41-nitrokey.rules /etc/udev/rules.d/
sudo udevadm control --reload-rules && sudo udevadm trigger


Then the instructions are misleading and don't include the important bits from https://github.com/linuxboot/heads/pull/1684#issuecomment-2122918284

```
user@heads-tests-deb12-nix-nk3:~$ nitropy nk3 update --version v1.7.1
Command line tool to interact with Nitrokey devices 0.4.47
Do you want to download the firmware version v1.7.1? [Y/n]: y
Download v1.7.1: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.03M/1.03M [00:00<00:00, 3.28MB/s]
Current firmware version:  v1.5.0
Updated firmware version:  v1.7.1

Please do not remove the Nitrokey 3 or insert any other Nitrokey 3 devices during the update. Doing so may damage the Nitrokey 3.
Do you want to perform the firmware update now? [y/N]: y

Please press the touch button to reboot the device into bootloader mode ...

Critical error:
No Nitrokey 3 bootloader device found

--------------------------------------------------------------------------------
Critical error occurred, exiting now
Unexpected? Is this a bug? Would you like to get support/help?
- You can report issues at: https://support.nitrokey.com/
- Writing an e-mail to support@nitrokey.com is also possible
- Please attach the log: '/tmp/nitropy.log.svejmix8' with any support/help request!
- Please check if you have udev rules installed: https://docs.nitrokey.com/nitrokey3/linux/firmware-update.html#troubleshooting

```
(changed mode of usb security dongle, needs repassing usb security dongle from qubes device widget)


```
user@heads-tests-deb12-nix-nk3:~$ nitropy nk3 update --version v1.7.1
Command line tool to interact with Nitrokey devices 0.4.47
Do you want to download the firmware version v1.7.1? [Y/n]: y
Download v1.7.1: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.03M/1.03M [00:00<00:00, 2.86MB/s]
Current firmware version:  [unknown]
Updated firmware version:  v1.7.1

Please do not remove the Nitrokey 3 or insert any other Nitrokey 3 devices during the update. Doing so may damage the Nitrokey 3.
Do you want to perform the firmware update now? [y/N]: y
Critical error:
Failed to perform firmware update
	Exception encountered: SPSDKConnectionError()

--------------------------------------------------------------------------------
Critical error occurred, exiting now
Unexpected? Is this a bug? Would you like to get support/help?
- You can report issues at: https://support.nitrokey.com/
- Writing an e-mail to support@nitrokey.com is also possible
- Please attach the log: '/tmp/nitropy.log.qwhw4rm1' with any support/help request!
- Please check if you have udev rules installed: https://docs.nitrokey.com/nitrokey3/linux/firmware-update.html#troubleshooting
```

That's where https://github.com/QubesOS/qubes-issues/issues/8953#issuecomment-1957154706 comes into play

nitropy nk3 reboot --bootloader 
(touch device)
(reassign nk3 to testing qube)
```
user@heads-tests-deb12-nix-nk3:~$ nitropy nk3 list
Command line tool to interact with Nitrokey devices 0.4.47
:: 'Nitrokey 3' keys
Critical error:
An unhandled exception occurred
	Exception encountered: SPSDKConnectionError()

--------------------------------------------------------------------------------
Critical error occurred, exiting now
Unexpected? Is this a bug? Would you like to get support/help?
- You can report issues at: https://support.nitrokey.com/
- Writing an e-mail to support@nitrokey.com is also possible
- Please attach the log: '/tmp/nitropy.log.72xhne1h' with any support/help request!
- Please check if you have udev rules installed: https://docs.nitrokey.com/nitrokey3/linux/firmware-update.html#troubleshooting
```

**Comment by tlaurion** [comment]
@daringer @jans23 provided proper feedback at https://github.com/QubesOS/qubes-issues/issues/8953#issuecomment-2129737616 and pinged @demimarie and @marmarek there and here to hopefully fix this upstream once and for all, this impacts not only nk3 but android debugging, android tethering and basically all usb passthrough outside of using everything in sys-usb which should not happen.

**Comment by tlaurion** [comment]
@mamarek @daringer  @demimarie the issue is that even if qvm-usb thinks that the device is passed to the destination qube, the qube itself doesn't see it, while standard tools cannot reset the device not the hub to force a redetection

Replication notes:
```
[user@dom0 ~]
$ qube=heads-tests-deb12-nix-nk3; device=$(qvm-usb | grep -e Nitrokey_Nitrokey_3 -e NXP_SEMICONDUCTOR_INC._USB_COMPOSITE_DEVICE | cut -d ' ' -f1); qvm-usb attach --persistent "$qube" "$device"
'device sys-usb:4-3 of class usb already attached to heads-tests-deb12-nix-nk3'
[user@dom0 ~]
(1)$ qvm-usb
BACKEND:DEVID  DESCRIPTION                                  USED BY
sys-usb:4-10   8087_0026                                    
sys-usb:4-3    NXP_SEMICONDUCTOR_INC._USB_COMPOSITE_DEVICE  heads-tests-deb12-nix-nk3
sys-usb:4-7    Sonix_Technology_Co.__Ltd._BisonCam_NB_Pro   
```

The problem here is that qubesos assigns sys-usb:4-3, not a pid-vid, and that the device not having been disconnected shows as if it was already passed to the qube, where

```
user@heads-tests-deb12-nix-nk3:~$ lsusb
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

Shows that the device is still under sys-usb control here, the control haven't been relinquished to the destination qube.
This is common issue for all usb devices passed by usb port instead of pid:vid.

**Comment by tlaurion** [comment]
Having users activate network under sys-usb and do this stuff under sys-usb instead of dedicated qube should not be recommended for QubesOS users. 

Ideally:
- [ ] QubesOS fixes qvm-usb+usb proxy with experimental packages so that sys-usb does the right thing

And the issue would be non-issue as well as all other usb composite devices having issues under QubesOS today.

@marmarek @demimarie @fepitre: Can this be prioritized?
---

Alternative/workaround if root issue cannot be fixed promptly:

- [ ] package nitrokeyapp2 for fedora-39
- [ ] package nitrokeyapp2 for debian-12
- [ ] have QubesOS distribute those packages so that instructions can be followed to install packages under sys-usb's template
- [ ] update documentation accordingly

**Comment by tlaurion** [comment]
@marmarek reproduced issue in nk3a nfc
https://github.com/QubesOS/qubes-issues/issues/8953#issuecomment-2135580836

File: nitrokey3/linux/firmware-update-qubes.rst (hunk 1 above)
Comment (nestire) [reply nitrokey3/linux/firmware-update-qubes.rst]: changed that

**Comment by tlaurion** [comment]
@nestire please add PR small description

**Pull request merged**